### PR TITLE
allow symfony 5

### PR DESCRIPTION
**Pull request by dbu**
fix #798

### CODE REVIEW

**Comment by SenseException** [comment]
Should 1.4 support Symfony 5?

**Comment by dbu** [comment]
> Should 1.4 support Symfony 5?

we have no 2.0 ready, and phpcr-odm only has loose ties to symfony, its just using the console component which does not change much between symfony 2 and 5.

**Comment by SenseException** [comment]
The builds of master and 1.4 are currently failing. How are PRs handled in phpcr-odm? The changes are looking good though, but it's hard to be sure if everything's all right for the repo.

**Comment by dbu** [comment]
i am trying to fix the build in this PR. by now its only a few versions left that make problems... once this is in 1.4 i'll try to update master with it.

the failing builds are mostly code rot. there is not much active development going on in this repo, and things that once worked have stopped working meanwhile for random reasons.

File: tests/Doctrine/Tests/ODM/PHPCR/Functional/ReorderTest.php [hunk 1]
```diff
@@ -102,7 +104,7 @@ public function testReorderNoop()
 
     public function testReorderNoObject()
     {
-        $this->setExpectedException('\Doctrine\ODM\PHPCR\Exception\InvalidArgumentException');
+        $this->expectException(InvalidArgumentException::class);
         $this->dm->reorder('parent', 'first', 'second', false);
         $this->dm->flush();
```
Comment (SenseException) [inline tests/Doctrine/Tests/ODM/PHPCR/Functional/ReorderTest.php]: I guess this `flush()` isn't needed here, but it's okay.

File: tests/Doctrine/Tests/ODM/PHPCR/Functional/Tools/Helper/TranslationConverterTest.php [hunk 2]
```diff
@@ -451,35 +454,32 @@ public function testTranslateStrict()
...
     }
 
-    /**
-     * @expectedException \Doctrine\ODM\PHPCR\Exception\InvalidArgumentException
-     * @expectedExceptionMessage To untranslate a document, you need to specify the previous translation strategy
-     */
     public function testUntranslateMissingPrevious()
     {
-        $class = 'Doctrine\Tests\Models\Blog\Comment';
+        $class = \Doctrine\Tests\Models\Blog\Comment::class;
```
Comment (SenseException) [inline tests/Doctrine/Tests/ODM/PHPCR/Functional/Tools/Helper/TranslationConverterTest.php]: `use` would be nice.

File: tests/Doctrine/Tests/ODM/PHPCR/Functional/Tools/Helper/TranslationConverterTest.php [hunk 3]
```diff
@@ -451,35 +454,32 @@ public function testTranslateStrict()
...
     }
 
-    /**
-     * @expectedException \Doctrine\ODM\PHPCR\Exception\InvalidArgumentException
-     * @expectedExceptionMessage need to specify the fields that where previously translated
-     */
     public function testUntranslateMissingFields()
     {
-        $class = 'Doctrine\Tests\Models\Blog\Comment';
+        $class = \Doctrine\Tests\Models\Blog\Comment::class;
```
Comment (SenseException) [inline tests/Doctrine/Tests/ODM/PHPCR/Functional/Tools/Helper/TranslationConverterTest.php]: `use` would be nice.

File: tests/Doctrine/Tests/ODM/PHPCR/Query/QueryTest.php [hunk 4]
```diff
@@ -184,22 +210,18 @@ public function testGetSingleResult_withOneResult()
...
-    /**
-     * @expectedException \Doctrine\ODM\PHPCR\Query\QueryException
-     */
     public function testGetSingleResult_withTwoResults()
     {
+        $this->expectException(QueryException::class);
         $this->phpcrQuery->expects($this->once())
             ->method('execute')
             ->will($this->returnValue(array('ok1', 'ok2')));
         $this->query->getSingleResult(Query::HYDRATE_PHPCR);
```
Comment (SenseException) [inline tests/Doctrine/Tests/ODM/PHPCR/Query/QueryTest.php]: This line might not needed?

File: tests/Doctrine/Tests/ODM/PHPCR/Translation/AttributeTranslationStrategyTest.php [hunk 5]
```diff
@@ -54,17 +63,24 @@ public function testGetLocalesFor()
...
+        $node = $this->createMock(NodeInterface::class);
         $properties = array();
 
         foreach (array_keys($localizedPropNames) as $localizedPropName) {
-            $property = $this->prophesize('PHPCR\PropertyInterface');
-            $property->getName()->willReturn($localizedPropName);
-            $properties[] = $property->reveal();
+            $property = $this->createMock(PropertyInterface::class);
+            $property
+                ->expects($this->any())
```
Comment (SenseException) [inline tests/Doctrine/Tests/ODM/PHPCR/Translation/AttributeTranslationStrategyTest.php]: `expects` isn't needed here.

File: tests/Doctrine/Tests/ODM/PHPCR/Functional/ReorderTest.php (hunk 1 above)
Comment (dbu) [reply tests/Doctrine/Tests/ODM/PHPCR/Functional/ReorderTest.php]: yeah i think you are right. there was a ton of those, where the test probably just was a copy of the happy path with an `@expectedException` annotation at the top, so it was hard to guess where it was expected to explode...

File: tests/Doctrine/Tests/ODM/PHPCR/Functional/Tools/Helper/TranslationConverterTest.php (hunk 3 above)
Comment (dbu) [reply tests/Doctrine/Tests/ODM/PHPCR/Functional/Tools/Helper/TranslationConverterTest.php]: there are two different comment classes used in this test :-/ i now made an alias for this one.

File: tests/Doctrine/Tests/ODM/PHPCR/Query/QueryTest.php (hunk 4 above)
Comment (dbu) [reply tests/Doctrine/Tests/ODM/PHPCR/Query/QueryTest.php]: the ordering is wrong. the line right after expect is setting up the partial mock to run the query.

**Comment by dbu** [comment]
given the amount of things i touch here and that i change what php versions we support and whatnot, i will make this version 1.5. i wait for this to build, then create the 1.5 branch rather than merging. i will change styleci to php 7.1 which should change the code to the nice array notation.

**Comment by dbu** [comment]
i pushed this as 1.5 branch

**Comment by dbu** [comment]
and thanks @SenseException for the review!

**Pull request closed without merging**